integration/src/main/java/scripts/testscripts/SnapshotDatasetOperations.java

```diff
@@ -0,0 +1,165 @@
+package scripts.testscripts;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import bio.terra.catalog.api.DatasetsApi;
+import bio.terra.catalog.client.ApiException;
+import bio.terra.catalog.model.CreateDatasetRequest;
+import bio.terra.catalog.model.StorageSystem;
+import bio.terra.datarepo.api.JobsApi;
+import bio.terra.datarepo.api.SnapshotsApi;
+import bio.terra.datarepo.model.JobModel;
+import bio.terra.datarepo.model.SnapshotRequestContentsModel;
+import bio.terra.datarepo.model.SnapshotRequestModel;
+import bio.terra.testrunner.runner.TestScript;
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import com.google.api.client.http.HttpStatusCodes;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scripts.client.CatalogClient;
+import scripts.client.DataRepoClient;
+
+/**
+ * A test for dataset operations using the catalog service endpoints, with TDR snapshots as the
+ * dataset source.
+ *
+ * <p>Create, Read, Update, Delete are tested.
+ */
+public class SnapshotDatasetOperations extends TestScript {
+
+  private static final Logger log = LoggerFactory.getLogger(SnapshotDatasetOperations.class);
+  private static final String TEST_DATASET_NAME = "CatalogTestDataset";
+
+  // TDR APIs
+  private SnapshotsApi snapshotsApi;
+  private UUID snapshotId;
+
+  // Catalog APis
+  private UUID datasetId;
+  private DatasetsApi datasetsApi;
+
+  @Override
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    DataRepoClient datarepoClient = new DataRepoClient(server, testUsers.get(0));
+    var datasetsApi = new bio.terra.datarepo.api.DatasetsApi(datarepoClient);
+    var dataset =
+        datasetsApi
+            .enumerateDatasets(null, null, null, null, TEST_DATASET_NAME, null)
+            .getItems()
+            .get(0);
+
+    snapshotsApi = new SnapshotsApi(datarepoClient);
+    var request =
+        new SnapshotRequestModel()
+            .name("catalog_integration_test_" + System.currentTimeMillis())
+            .description("catalog test snapshot")
+            .profileId(dataset.getDefaultProfileId())
+            .addContentsItem(
+                new SnapshotRequestContentsModel()
+                    .datasetName(TEST_DATASET_NAME)
+                    .mode(SnapshotRequestContentsModel.ModeEnum.BYFULLVIEW));
+    var createJobId = snapshotsApi.createSnapshot(request).getId();
+
+    var jobApi = new JobsApi(datarepoClient);
+    JobModel job;
```

Comment:
Should this explicitly have the `jobStatus` set to RUNNING? Right now that is the default, but just curious if that could change.

Reply:
The code doesn't look at `job` until after `jobApi.retrieveJob(createJobId)` is called. So the contents of `job` at this point (which is `null`) doesn't matter.

it's true that if the Job API adds a new state, like `IN_QUEUE` or `PENDING` then this code could break. In that case maybe checking for `... == SUCCEEDED || .. == FAILED` would make more sense. But that could break too if another end state like `ABORTED` is added. My thinking is that if this does change in the future, the test will fail and then we'll have to update the code to match the current API behavior, so I wouldn't worry too much about planning for change now.